pandas/tests/extension/test_boolean.py

```diff
@@ -367,6 +372,30 @@ def check_reduce(self, s, op_name, skipna):
             expected = bool(expected)
         tm.assert_almost_equal(result, expected)
 
+    def check_reduce_and_wrap(self, ser: pd.Series, op_name: str, skipna: bool):
+        if op_name in ["count", "kurt", "sem"]:
+            pytest.skip(f"{op_name} not an array method")
+
+        arr = ser.array
+
+        if op_name in ["mean", "median", "var", "std", "skew"]:
+            cmp_dtype = "Float64"
+        elif op_name in ["min", "max"]:
+            cmp_dtype = "boolean"
+        elif op_name in ["sum", "prod"]:
+            is_windows_or_32bit = is_platform_windows() or not IS64
+            cmp_dtype = "Int32" if skipna and is_windows_or_32bit else "Int64"
```

Comment:
I've had a big struggle to get a handle on how the dtype end up changing on different systems, so this is definitely not deliberate on my part, just trying to get the tests to pass, then it will be easier to fix up whatever needs to be fixed.

But I think you are probably right, I can look into it tomorrow. Having said that, I don't think I've changed anything in this regard, so if this is a bug, this just uncovers an existing bug and it shouldn't be not something that has been introduced in this PR.

Reply:
I've fixed this, so skipna doesn't affect the dtype.

Also: "I don't think I've changed anything in this regard." are always famous last words. My PR was the cause of this, so good catch.